crypto/stateproof/structs.go

```diff
@@ -23,35 +23,40 @@ import (
 	"github.com/algorand/go-algorand/data/basics"
 )
 
-// Params defines common parameters for the verifier and builder.
-type Params struct {
-	Msg          crypto.Hashable // Message to be certified
-	ProvenWeight uint64          // Weight threshold proven by the certificate
-	SigRound     basics.Round    // The round for which the ephemeral key is committed to
-	SecKQ        uint64          // Security parameter (k+q) from analysis document
-}
+// MessageHash represents the message that a state proof will attest to.
+type MessageHash [32]byte
 
-// CompactOneTimeSignature is crypto.OneTimeSignature with omitempty
-type CompactOneTimeSignature struct {
-	_struct struct{} `codec:",omitempty,omitemptyarray"`
-	merklesignature.Signature
+// MessageHashType is the type of hash used to generate MessageHash
+const MessageHashType = crypto.Sha256
```

Comment:
can be moved to const

Reply:
Can actually be removed altogether since it's only used in one place: `Message.IntoStateProofMessageHash()`